hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/SigTransactionMetadataBuilder.java

```diff
@@ -104,14 +133,16 @@ public void addNonPayerKey(final AccountID id) {
      * failureReason is null, sets the default failureReason given in the result.
      *
      * @param id given accountId
-     * @param failureStatus given failure status
+     * @param failureStatusToUse failure status to be set if there is failure
      */
-    public void addNonPayerKey(final AccountID id, final ResponseCodeEnum failureStatus) {
+    public SigTransactionMetadataBuilder addNonPayerKey(
+            final AccountID id, final ResponseCodeEnum failureStatusToUse) {
```

Comment:
```suggestion
            @NonNull final AccountID id, @NonNull final ResponseCodeEnum failureStatusToUse) {
```

Reply:
This needs `Nullable` annotation for `failureStatusToUse`. If the status is given null, we use default `INVALID_ACCOUNT_ID` if there is a failure.